libraries/AP_KDECAN/AP_KDECAN.cpp

```diff
@@ -158,7 +158,7 @@ void AP_KDECAN::init(uint8_t driver_index, bool enable_filters)
         debug_can(2, "KDECAN: found ESC id %u\n\r", id.source_id);
     }
 
-    snprintf(_thread_name, sizeof(_thread_name), "kdecan_%u", driver_index);
+    snprintf(_thread_name, sizeof(_thread_name), "kdecan_%u", driver_index & MAX_NUMBER_OF_CAN_DRIVERS);
```

Comment:
this doesn't work. You should use the mask of bits lower than the last bit MAX_NUMBER_OF_CAN_DRIVERS. If that number is guaranteed to be power of 2, it's simply `(MAX_NUMBER_OF_CAN_DRIVERS << 1) - 1`, otherwise you need to use a instruction to find the most significant bit, like `__builtin_clz()`. Not worth IMO, just increase the _thread_name storage to 16.